reference/rest/src/main/java/io/a2a/server/apps/quarkus/A2AServerRoutes.java

```diff
@@ -0,0 +1,416 @@
+package io.a2a.server.apps.quarkus;
+
+import static io.vertx.core.http.HttpHeaders.CONTENT_TYPE;
+import static jakarta.ws.rs.core.MediaType.APPLICATION_JSON;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Flow;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import io.a2a.server.ServerCallContext;
+import io.a2a.server.auth.UnauthenticatedUser;
+import io.a2a.server.auth.User;
+import io.a2a.server.util.async.Internal;
+import io.a2a.spec.AgentCard;
+import io.a2a.spec.InternalError;
+import io.a2a.spec.JSONRPCError;
+import io.a2a.transport.rest.handler.RestHandler;
+import io.a2a.transport.rest.handler.RestHandler.HTTPRestResponse;
+import io.a2a.transport.rest.handler.RestHandler.HTTPRestStreamingResponse;
+import io.quarkus.logging.Log;
+import io.quarkus.vertx.web.Body;
+import io.quarkus.vertx.web.ReactiveRoutes;
+import io.quarkus.vertx.web.Route;
+import io.smallrye.mutiny.Multi;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+@Singleton
+public class A2AServerRoutes {
+
+    @Inject
+    RestHandler jsonRestHandler;
+
+    // Hook so testing can wait until the MultiSseSupport is subscribed.
+    // Without this we get intermittent failures
+    private static volatile Runnable streamingMultiSseSupportSubscribedRunnable;
+
+    @Inject
+    @Internal
+    Executor executor;
+
+    @Inject
+    Instance<CallContextFactory> callContextFactory;
+
+    @Route(regex = "^/v1/message:send$", methods = {Route.HttpMethod.POST}, consumes = {APPLICATION_JSON}, type = Route.HandlerType.BLOCKING)
+    public void sendMessage(@Body String body, RoutingContext rc) {
+        ServerCallContext context = createCallContext(rc);
+        HTTPRestResponse response = null;
+        try {
+            response = jsonRestHandler.sendMessage(body, context);
+        } catch (Throwable t) {
+            if (t instanceof JSONRPCError error) {
+                response = jsonRestHandler.createErrorResponse(error);
```

Comment:
Just to check, any reason to catch the JSONRPCErrors in this class as opposed to in the RestHandler? Just wondering if the call to `jsonRestHandler.sendMessage` could return an appropriate response even in the error case. That way, all we need to do here is catch any other throwable and wrap it with an InternalError just to be safe.

Reply:
e.g, in the JSONRPCHandler, we have this error handling:

https://github.com/a2aproject/a2a-java/blob/main/transport/jsonrpc/src/main/java/io/a2a/transport/jsonrpc/handler/JSONRPCHandler.java#L71-L79